x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java

```diff
@@ -145,34 +152,70 @@ public void init() throws Exception {
+        commonFields = new LoggingAuditTrail.EntryCommonFields(settings, localNode).commonFields;
         threadContext = new ThreadContext(Settings.EMPTY);
         if (randomBoolean()) {
-            String id = randomAlphaOfLength(10);
-            threadContext.putHeader(Task.X_OPAQUE_ID, id);
-            opaqueId = ", opaque_id=[" + id + "]";
-        } else {
-            opaqueId = "";
+            threadContext.putHeader(Task.X_OPAQUE_ID, randomAlphaOfLengthBetween(1, 4));
         }
+        patternLayout = PatternLayout.newBuilder().withPattern(
+                "{" +
+                "\"timestamp\":\"%d{ISO8601}\"" +
+                "%varsNotEmpty{, \"node.name\":\"%enc{%map{node.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"host.name\":\"%enc{%map{host.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"host.ip\":\"%enc{%map{host.ip}}{JSON}\"}" +
+                "%varsNotEmpty{, \"event.type\":\"%enc{%map{event.type}}{JSON}\"}" +
+                "%varsNotEmpty{, \"event.action\":\"%enc{%map{event.action}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.name\":\"%enc{%map{user.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.run_by.name\":\"%enc{%map{user.run_by.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.run_as.name\":\"%enc{%map{user.run_as.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.realm\":\"%enc{%map{user.realm}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.run_by.realm\":\"%enc{%map{user.run_by.realm}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.run_as.realm\":\"%enc{%map{user.run_as.realm}}{JSON}\"}" +
+                "%varsNotEmpty{, \"user.roles\":\"%enc{%map{user.roles}}{JSON}\"}" +
+                "%varsNotEmpty{, \"origin.type\":\"%enc{%map{origin.type}}{JSON}\"}" +
+                "%varsNotEmpty{, \"origin.address\":\"%enc{%map{origin.address}}{JSON}\"}" +
+                "%varsNotEmpty{, \"realm\":\"%enc{%map{realm}}{JSON}\"}" +
+                "%varsNotEmpty{, \"url.path\":\"%enc{%map{url.path}}{JSON}\"}" +
+                "%varsNotEmpty{, \"url.query\":\"%enc{%map{url.query}}{JSON}\"}" +
+                "%varsNotEmpty{, \"request.body\":\"%enc{%map{request.body}}{JSON}\"}" +
+                "%varsNotEmpty{, \"action\":\"%enc{%map{action}}{JSON}\"}" +
+                "%varsNotEmpty{, \"request.name\":\"%enc{%map{request.name}}{JSON}\"}" +
+                "%varsNotEmpty{, \"indices\":\"%enc{%map{indices}}{JSON}\"}" +
+                "%varsNotEmpty{, \"opaque_id\":\"%enc{%map{opaque_id}}{JSON}\"}" +
+                "%varsNotEmpty{, \"transport.profile\":\"%enc{%map{transport.profile}}{JSON}\"}" +
+                "%varsNotEmpty{, \"rule\":\"%enc{%map{rule}}{JSON}\"}" +
+                "%varsNotEmpty{, \"event.category\":\"%enc{%map{event.category}}{JSON}\"}" +
+                "}%n")
+                .withCharset(StandardCharsets.UTF_8)
+                .build();
```

Comment:
Duplicate the pattern from `log4j2.properties` here. Not strictly necessary to have the same format but a different one is unwarranted.